consensus/src/pipeline/buffer_item.rs

```diff
@@ -38,47 +44,18 @@ fn generate_commit_ledger_info(
     )
 }
 
-fn verify_signatures(
-    unverified_signatures: PartialSignatures,
-    validator: &ValidatorVerifier,
+fn ledger_info_with_unverified_signatures(
+    unverified_votes: HashMap<Author, CommitVote>,
     commit_ledger_info: &LedgerInfo,
-) -> PartialSignatures {
-    // Returns a valid partial signature from a set of unverified signatures.
-    // TODO: Validating individual signatures in expensive. Replace this with optimistic signature
-    // verification for BLS. Here, we can implement a tree-based batch verification technique that
-    // filters out invalid signature shares much faster when there are only a few of them
-    // (e.g., [LM07]: Finding Invalid Signatures in Pairing-Based Batches,
-    // by Law, Laurie and Matt, Brian J., in Cryptography and Coding, 2007).
-    PartialSignatures::new(
-        unverified_signatures
-            .signatures()
-            .iter()
-            .filter(|(author, sig)| validator.verify(**author, commit_ledger_info, sig).is_ok())
-            .map(|(author, sig)| (*author, sig.clone()))
-            .collect(),
-    )
-}
-
-fn generate_executed_item_from_ordered(
-    commit_info: BlockInfo,
-    executed_blocks: Vec<PipelinedBlock>,
-    verified_signatures: PartialSignatures,
-    callback: StateComputerCommitCallBackType,
-    ordered_proof: LedgerInfoWithSignatures,
-    order_vote_enabled: bool,
-) -> BufferItem {
-    debug!("{} advance to executed from ordered", commit_info);
-    let partial_commit_proof = LedgerInfoWithVerifiedSignatures::new(
-        generate_commit_ledger_info(&commit_info, &ordered_proof, order_vote_enabled),
-        verified_signatures,
-    );
-    BufferItem::Executed(Box::new(ExecutedItem {
-        executed_blocks,
-        partial_commit_proof,
-        callback,
-        commit_info,
-        ordered_proof,
-    }))
+) -> LedgerInfoWithUnverifiedSignatures {
+    let mut li_with_sig = LedgerInfoWithUnverifiedSignatures::new(commit_ledger_info.clone());
+    for vote in unverified_votes.values() {
+        let sig = vote.signature_with_status();
+        if vote.ledger_info() == commit_ledger_info {
+            li_with_sig.add_signature(vote.author(), sig);
```

Comment:
nit: this function signature looks awkward, it should take a ownership of sig instead of a reference and clone inside